pywr/parameters/_parameters.pyx

```diff
@@ -905,6 +906,109 @@ cdef class UniformDrawdownProfileParameter(Parameter):
+    """Parameter which interpolates a daily profile using a radial basis function (RBF).
+
+    The daily profile is computed during model `reset` using a radial basis function with
+    day-of-year as the independent variables. The days of the year are defined by the user
+    alongside the values to use on each of those days for the interpolation. The first
+    day of the years should always be one, and its value is repeated as the 365th value.
+    In addition the second and penultimate values are mirrored to encourage a consistent
+    gradient to appear across the boundary. The RBF calculations are undertaken using
+    the `scipy.interpolate.Rbf` object, please refer to Scipy's documentation for more
+    information.
+
+    Parameters
+    ----------
+    days_of_year : iterable, integer
+        The days of the year at which the interpolation values are defined. The first
+        value should be one.
+    values : iterable, float
+        Values to use for interpolation corresponding to the `days_of_year`.
+    lower_bounds : float (default=0.0)
+        The lower bounds of the values when used during optimisation.
+    upper_bounds : float (default=np.inf)
+        The upper bounds of the values when used during optimisation.
+    rbf_kwargs: Optional, dict
+        Optional dictionary of keyword arguments to base to the Rbf object.
+
+    """
+    def __init__(self, model, days_of_year, values, lower_bounds=0.0, upper_bounds=np.inf, rbf_kwargs=None, **kwargs):
+        super(RbfProfileParameter, self).__init__(model, **kwargs)
+
+        if len(days_of_year) != len(values):
+            raise ValueError(f"The length of values ({len(values)}) must equal the length of "
+                             f"`days_of_year` ({len(days_of_year)}).")
+
+        self.double_size = len(values)
+        self.integer_size = 0
+        self._values = np.array(values)
+        self._days_of_year = np.array(days_of_year, dtype=np.int32)
+        self._lower_bounds = np.ones(self.double_size)*lower_bounds
+        self._upper_bounds = np.ones(self.double_size)*upper_bounds
+        self.rbf_kwargs = rbf_kwargs if rbf_kwargs is not None else {}
+
+    cpdef reset(self):
+        Parameter.reset(self)
+        # The interpolated profile is recalculated during reset so that
+        # it will update when the _values array is updated via `set_double_variables`
+        # and the model is rerun. I.e. during optimisation (where setup is not redone).
+        self._interpolate()
+
+    cpdef _interpolate(self):
+        cdef int i
+
+        days_of_year = list(self._days_of_year)
+        # Append day 365 to the list and mirror the penultimate and second DOY at the start and end
+        # of the list respectively. This helps ensure the gradient is roughly the same across the boundary
+        # between days 365 and 0.
+        days_of_year = [days_of_year[-1]-365+1] + list(days_of_year) + [365, 365+days_of_year[1]-1]
```

Comment:
If one of the input 'days_of_year' is 365 then adding another one here will cause an error. probably needs a check/warning in the docstring/init

Reply:
I think this is incorrect actually. It should be repeating day 1 as value 366 - I'm assuming zero based indexing here.

Reply:
yeah ok, forgot about the zero indexing!

Reply:
This should be fixed now. I've added some checking of the days of the year.